lib/view-models/search-view-model.coffee

```diff
@@ -30,5 +30,5 @@ class SearchViewModel extends ViewModel
       @restoreHistory(@historyIndex)
 
   confirm: (view) =>
-    @vimState.pushSearchHistory(@)
```

Comment:
Yeah, I'll definitely merge this. Do you mind if I do so now, and you can do the features you mentioned above on a separate PR?